beacon-chain/blockchain/service.go

```diff
@@ -431,6 +431,12 @@ func (s *Service) initializeChainInfo(ctx context.Context) error {
 		if err != nil {
 			return errors.Wrap(err, "could not hash head block")
 		}
+		finalizedState, err := s.stateGen.Resume(ctx)
```

Comment:
Is there a bug that caused by not running `s.stateGen.Resume` here?

If yes, can you please add some descriptions for the bug? and maybe a regression test?

Reply:
There is no bug, the finalized state is accessed for info purposes.

Reply:
Ah not really I just saw that we resumed stategen in the normal path, so I thought it would be correct to do it here too. I used the archived state to determine the starting point of our head sync.